pkg/controllers/allocation/controller.go

```diff
@@ -92,18 +106,60 @@ func (c *Controller) Reconcile(ctx context.Context, object client.Object) (recon
 		packings = append(packings, c.packer.Pack(ctx, constraintGroup, instanceTypes)...)
 	}
 
-	// 4. Create packedNodes for packings
+	// 6. Create packedNodes for packings
 	packedNodes, err := c.cloudProvider.Create(ctx, provisioner, packings)
 	if err != nil {
 		return reconcile.Result{}, fmt.Errorf("creating capacity, %w", err)
 	}
 
-	// 5. Bind pods to nodes
+	// 7. Bind pods to nodes
+	var errs error
 	for _, packedNode := range packedNodes {
 		zap.S().Infof("Binding pods %v to node %s", apiobject.PodNamespacedNames(packedNode.Pods), packedNode.Node.Name)
 		if err := c.binder.Bind(ctx, packedNode.Node, packedNode.Pods); err != nil {
 			zap.S().Errorf("Continuing after failing to bind, %s", err.Error())
+			errs = multierr.Append(errs, err)
 		}
 	}
-	return reconcile.Result{RequeueAfter: c.Interval()}, nil
+	return reconcile.Result{}, errs
+}
+
+func (c *Controller) podToProvisioner(o client.Object) (requests []reconcile.Request) {
+	pod := o.(*v1.Pod)
+	ctx := context.Background()
+	provisioner, err := c.filter.GetProvisionerFor(ctx, pod)
+	if err != nil {
+		zap.S().Errorf("Retrieving provisioner, %s", err.Error())
+		return nil
+	}
+	if err = c.filter.isProvisionable(ctx, pod, provisioner); err != nil {
+		return nil
+	}
+	c.batcher.Add(provisioner)
+	return []reconcile.Request{{NamespacedName: types.NamespacedName{Name: provisioner.Name, Namespace: provisioner.Namespace}}}
+}
+
+func (c *Controller) Register(ctx context.Context, m manager.Manager) error {
+	err := controllerruntime.
+		NewControllerManagedBy(m).
+		Named("Allocation").
+		For(&v1alpha2.Provisioner{}).
+		Watches(
+			// Reconcile all pods related to a provisioner when it changes.
+			&source.Kind{Type: &v1.Pod{}},
+			handler.EnqueueRequestsFromMapFunc(c.podToProvisioner),
+		).
+		WithOptions(
```

Comment:
Remind me why we can't use controller-runtime's defaults? Is it too fast?

Reply:
Not sure, @njtran had tuned these recently, so I just grabbed those params. Do you know off-hand @njtran ?